airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -344,13 +347,10 @@ public List<Job> listJobs(final ConfigType configType, final String configId, fi
 
   @Override
   public List<Job> listJobs(final Set<ConfigType> configTypes, final String configId, final int pagesize, final int offset) throws IOException {
+    final String jobsSubquery = "(SELECT * FROM jobs WHERE CAST(jobs.config_type AS VARCHAR) in " + Sqls.toSqlInFragment(configTypes)
+        + " AND jobs.scope = '" + configId + "' ORDER BY jobs.created_at DESC, jobs.id DESC LIMIT " + pagesize + " OFFSET " + offset + ") AS jobs";
```

Comment:
A future warning for us:  this is fine for now, but if/when the dataset grows large, OFFSET can have some real performance issues.  The alternative is to filter using a key or ID, but that has its own problems.  This also highlights the need to use pagination from an application framework so we can avoid having to solve this type of problem (e.g. let the framework handle how to generate a paginated query).

Reply:
Yep, I definitely agree that using an application framework to generate these queries for us would be greatly preferable. Looks like we also have an EPIC for API Pagination [here](https://github.com/airbytehq/airbyte/issues/14282) - it may be a good idea for you to add a note about avoiding OFFSET and doing a different pagination mechanism (e.g. cursor pagination)